step-templates/octopus-register-listening-worker.json

```diff
@@ -0,0 +1,94 @@
m.Name -eq $RegistrationName)\n    {\n    \t$machineFound = $true\n        break\n    }\n}\n\nif ($machineFound)\n{\n\tWrite-Highlight \"Machine already exists, skipping registration\"\n    Exit 0\n}\n\n$workerPoolList = $WorkerPools -split \",\"\n$workerPoolIdList = @()\nWrite-Host \"Getting the ids for all environments specified\"\nforeach($workerPool in $workerPoolList)\n{\n\tWrite-Host \"Getting the id for the worker pool $workerPool\"\n    \n    if ($workerPool.StartsWith(\"WorkerPools-\") -eq $true)\n    {\n    \tWrite-Host \"The worker pool is already an id, using that instead of looking it up\"\n    \t$workerPoolIdList += $workerPool\n    }\n    else\n    {\n    \t$workerPoolEscaped = $workerPool.Replace(\" \", \"%20\")\n        $workerPoolUrl = \"$OctopusUrl/api/$SpaceId/workerpools?skip=0&take=1000&partialName=$workerPoolEscaped\"\n        $workerPoolResponse = Invoke-RestMethod $workerPoolUrl -Headers $header \n\n        $workerPoolId = $workerPoolResponse.Items[0].Id\n        Write-Host \"The id for worker pool $workerPool is $workerPoolId\"\n        $workerPoolIdList += $workerPoolId\n    }       \n}\n\n$machinePolicyId = $machinePolicyIdOrName\nif ($machinePolicyIdOrName.StartsWith(\"MachinePolicies-\") -eq $false)\n{\n\tWrite-Host \"The machine policy specified $machinePolicyIdOrName appears to be a name\"\n\t$machinePolicyNameEscaped = $machinePolicyIdOrName.Replace(\" \", \"%20\")\n\t$machinePolicyResponse = Invoke-RestMethod \"$OctopusUrl/api/$SpaceId/machinepolicies?partialName=$machinePolicyNameEscaped\" -Headers $header\n        \n    $machinePolicyId = $machinePolicyResponse.Items[0].Id\n    Write-Host \"The machine policy id is $machinePolicyId\"\n}\n\n$discoverUrl = \"$OctopusUrl/api/$SpaceId/machines/discover?host=$RegistrationAddress&port=$PortNumber&type=TentaclePassive\"\nWrite-Host \"Discovering the machine $discoverUrl\"\n$discoverResponse = Invoke-RestMethod $discoverUrl -Headers $header \nWrite-Host \"ProjectResponse: $discoverResponse\"\n\n$machineThumbprint = $discoverResponse.EndPoint.Thumbprint\nWrite-Host \"Thumbprint = $machineThumbprint\"\n\n$rawRequest = @{\n  Id = $null;\n  MachinePolicyId = $MachinePolicyId;\n  Name = $RegistrationName;\n  IsDisabled = $false;\n  HealthStatus = \"Unknown\";\n  HasLatestCalamari = $true;\n  StatusSummary = $null;\n  IsInProcess = $true;\n  Links = $null;\n  WorkerPoolIds = $workerPoolIdList;\n  Endpoint = @{\n    Id = $null;\n    CommunicationStyle = \"TentaclePassive\";\n    Links = $null;\n    Uri = \"https://$RegistrationAddress`:$PortNumber\";\n    Thumbprint = \"$machineThumbprint\";\n    ProxyId = $null\n  }\n}\n\n$jsonRequest = $rawRequest | ConvertTo-Json -Depth 10\n\nWrite-Host \"Sending in the request $jsonRequest\"\n\n$machineUrl = \"$OctopusUrl/api/$SpaceId/workers\"\nWrite-Host \"Creating the worker\"\n$machineResponse = Invoke-RestMethod $machineUrl -Headers $header -Method POST -Body $jsonRequest\n\nWrite-Host \"Create workers's response: $machineResponse\""
```

Comment:
Same comments re: space id and overwrite

Reply:
Added space check